pkg/reconciler/v1alpha1/taskrun/taskrun_test.go

```diff
@@ -217,21 +283,57 @@ func TestReconcileBuildsCreated(t *testing.T) {
 				t.Errorf("expected no error. Got error %v", err)
 			}
 
-			if len(client.bclient.Actions()) == 0 {
+			if len(client.buildClient.Actions()) == 0 {
 				t.Errorf("Expected actions to be logged in the buildclient, got none")
 			}
 			namespace, name, err := cache.SplitMetaNamespaceKey(tc.taskRun)
 			if err != nil {
 				t.Errorf("Invalid resource key: %v", err)
 			}
 			// check error
-			build, err := client.bclient.BuildV1alpha1().Builds(namespace).Get(name, metav1.GetOptions{})
+			build, err := client.buildClient.BuildV1alpha1().Builds(namespace).Get(name, metav1.GetOptions{})
 			if err != nil {
 				t.Errorf("Failed to fetch build: %v", err)
 			}
-			if d := cmp.Diff(build.Spec, tc.wantedBuildSpec); d != "" {
+			if d := cmp.Diff(build.Spec, tc.expectedBuildSpec, cmp.AllowUnexported(corev1.PersistentVolumeClaim{})); d != "" {
 				t.Errorf("buildspec doesn't match, diff: %s", d)
 			}
+
+			//Command, Args, Env, VolumeMounts
+			if len(client.kubeClient.Actions()) == 0 {
+				t.Fatalf("Expected actions to be logged in the kubeclient, got none")
+			}
+			// 3. check that volume was created
+			pvc, err := client.kubeClient.CoreV1().PersistentVolumeClaims(namespace).Get(name, metav1.GetOptions{})
+			if err != nil {
+				t.Errorf("Failed to fetch build: %v", err)
+			}
+
+			// get related TaskRun to populate expected PVC
+			tr, err := client.pipelineClient.PipelineV1alpha1().TaskRuns(namespace).Get(name, metav1.GetOptions{})
+			if err != nil {
+				t.Errorf("Failed to fetch build: %v", err)
+			}
+			expectedVolume := GetExpectedVolume(tr)
+			if d := cmp.Diff(pvc.Name, expectedVolume.Name); d != "" {
+				t.Errorf("pvc doesn't match, diff: %s", d)
+			}
+			if d := cmp.Diff(pvc.OwnerReferences, expectedVolume.OwnerReferences); d != "" {
+				t.Errorf("pvc doesn't match, diff: %s", d)
+			}
+			if d := cmp.Diff(pvc.Spec.AccessModes, expectedVolume.Spec.AccessModes); d != "" {
+				t.Errorf("pvc doesn't match, diff: %s", d)
+			}
+			// TODO(aaron-prindle) make these tests verify exact pvc spec, not just name
+
+			// cannot verify pvc.Spec.Resources currently, getting error:
+			// panic: cannot handle unexported field: {v1.ResourceRequirements}.Requests["storage"].i
+			// consider using AllowUnexported or cmpopts.IgnoreUnexported [recovered]
+
+			// if d := cmp.Diff(pvc.Spec.Resources, expectedVolume.Spec.Resources); d != "" {
```

Comment:
you could assert on the individual fields in `Resources` - `cmp.Diff` is probably trying to assert on every single field in the object, including fields which aren't exposed.

you just need to look at the values we set in:
![image](https://user-images.githubusercontent.com/432502/47321961-62d39400-d624-11e8-8776-b19966ec6f46.png)

Reply:
done